src/store/concepts/session/operations/login.js

```diff
@@ -0,0 +1,58 @@
+import { createLogic } from 'redux-logic'
+import Cookie from 'js-cookie'
+
+import { SUBMIT_USER_LOGIN } from 'Store/concepts/session/types'
+import {
+  requestToken,
+  newSession,
+  validateRequestToken
+} from 'Store/concepts/session/endpoints'
+import { changeLoggedStatus } from 'Store/concepts/session/actions'
+
+const loginUserLogicOperation = createLogic({
+  type: SUBMIT_USER_LOGIN,
+  latest: true,
+  async process(
+    {
+      action: {
+        values: { username, password }
+      },
+      httpClient
+    },
+    dispatch,
+    done
+  ) {
+    try {
+      /* eslint-disable camelcase */
+      const {
+        data: { request_token }
+      } = await httpClient.get(requestToken)
+      await httpClient.post(validateRequestToken, {
+        username,
+        password,
+        request_token
+      })
+      const {
+        data: { session_id }
+      } = await httpClient.post(newSession, { request_token })
+      /* eslint-enable camelcase */
+
+      Cookie.set('session_id', session_id)
+
+      dispatch(changeLoggedStatus(true))
+    } catch (e) {
+      if (e.status === 401) {
+        // eslint-disable-next-line no-console
+        console.log('Username or password are incorrect')
+        return done()
```

Comment:
useless line

Reply:
Why unless? If we have specific status code, we have special error and action complete now, but I remove it, I should add else block to next part

Reply:
you already have  `done()` outside of `try...catch` block, so `done()` will be called in any case